### ISSUE TITLE
ValueError when using Equal and NotEqual primitives

### ISSUE THREAD

**Issue by thehomebrewnerd** [FeatureLabs/featuretools#845]
#### Bug Description

When running DFS with the `Equal` and `NotEqual` primitives on the featuretools ecommerce entityset (target entity = customers), the following error message is generated:
`ValueError: When changing to a larger dtype, its size must be a divisor of the total size in bytes of the last axis of the array.`

This can error be duplicated with this code:
```
from featuretools.tests.testing_utils import make_ecommerce_entityset
es = make_ecommerce_entityset()
trans_primitives = ['equal']
fm, _ = ft.dfs(entityset=es, target_entity='customers', trans_primitives=trans_primitives)
```

#### Output of ``featuretools.show_info()``

<details>

Featuretools version: 0.14.0.dev0

SYSTEM INFO
-----------
python: 3.7.4.final.0
python-bits: 64
OS: Darwin
OS-release: 18.7.0
machine: x86_64
processor: i386
byteorder: little
LC_ALL: None
LANG: en_US.UTF-8
LOCALE: en_US.UTF-8

INSTALLED VERSIONS
------------------
numpy: 1.17.3
pandas: 0.25.1
tqdm: 4.36.1
PyYAML: 5.1.2
cloudpickle: 1.2.2
dask: 2.6.0
distributed: 2.6.0
psutil: 5.6.3
Click: 7.0
pip: 19.3
setuptools: 40.8.0

</details>

**Comment by rwedge** [comment]
It looks like this error is happening when trying to apply the Equal primitive to a Boolean feature and a Datetime feature.  Currently the `Equal` primitive input types are `[Variable, Variable]`, which allows for any variable type for either input. 

One idea for addressing this would be to limit `Equal` to apply to matching variables in some way.  We don't support that kind of expression with input types right now, so this would require adding new functionality.

**Comment by rwedge** [comment]
Fixed in v0.15.0 release
